synapseclient/core/credentials/credential_provider.py

```diff
@@ -145,11 +151,44 @@ def _get_auth_info(self, syn, user_login_args):
         return username, password, api_key, auth_token
 
 
+class AWSParameterStoreCredentialsProvider(SynapseCredentialsProvider):
+    """
+    Retrieves user's authentication token from AWS SSM Parameter store
+    """
+
+    ENVIRONMENT_VAR_NAME = "SYNAPSE_TOKEN_AWS_SSM_PARAMETER_NAME"
+
+    def _get_auth_info(self, syn, user_login_args):
+        ssm_param_name = os.environ.get(self.ENVIRONMENT_VAR_NAME)
+        token = None
+        if ssm_param_name:
+            boto3 = utils.attempt_import("boto3",
+                                         "\n\nThe Synapse client uses boto3 "
+                                         "in order to access Systems Manager Parameter Storage.\n")
+            import botocore  # if can import boto3, can import botocore
+            try:
+                ssm_client = boto3.client('ssm')
+                result = ssm_client.get_parameter(
+                    Name=ssm_param_name,
+                    WithDecryption=True,
+                )
+                token = result['Parameter']['Value']
+            except botocore.exceptions.ClientError as e:
+                syn.logger.warn(f'{self.ENVIRONMENT_VAR_NAME} was defined as {ssm_param_name}, '
```

Comment:
Is every ClientError here a missing SSM param? Also this will only log the ClientError message and not the stack. I'm ambivalent whether the full stack is useful here but if so e.g. syn.logger.warn(<message>, exc_info=True)